pkg/client/database/postgres/postgres.go

```diff
@@ -0,0 +1,294 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	"strings"
+
+	"github.com/crossplane-contrib/provider-in-cluster/pkg/controller/utils"
+	"github.com/google/uuid"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/crossplane-contrib/provider-in-cluster/apis/database/v1alpha1"
+)
+
+const (
+	errGetPasswordSecretFailed = "cannot get password secret"
+	NamespacePrefixOpenShift   = "openshift-"
+	DefaultPostgresPort        = 5432
+)
+
+type Client interface {
+	CreateOrUpdate(ctx context.Context, obj runtime.Object) (controllerutil.OperationResult, error)
+	ParseInputSecret(ctx context.Context, postgres v1alpha1.Postgres) (string, error)
+	DeleteBucketPVC(ctx context.Context, postgres *v1alpha1.Postgres) error
```

Comment:
Bucket? should these be renamed lines 49 to 51

Reply:
Fixed, thanks for the catch